starfish/core/image/_filter/reduce.py

```diff
@@ -0,0 +1,123 @@
+from starfish.core.types import Axes
+from starfish.core.util import click
+from ._base import FilterAlgorithmBase
+
+
+class Reduce(FilterAlgorithmBase):
+    """
+    Reduces the dimensions of the ImageStack by applying a function
+    along one or more axes.
+
+    Parameters
+    ----------
+    dims : Axes
+        one or more Axes to project over
+    func : Union[str, Callable]
+        function to apply across the dimension(s) specified by dims.
+        If a functino is provided, it follow the form specified by
+        DataArray.reduce():
+        http://xarray.pydata.org/en/stable/generated/xarray.DataArray.reduce.html
+        The following strings are valid:
+            max: maximum intensity projection (applies numpy.amax)
+            mean: take the mean across the dim(s) (applies numpy.mean)
+            sum: sum across the dim(s) (applies numpy.sum)
+
+    See Also
+    --------
+    starfish.types.Axes
+
+    """
+
+    def __init__(
+        self, dims: Iterable[Union[Axes, str]], func: Union[str, Callable] = 'max'
+    ) -> None:
+
+        self.dims = dims
+
+        # If the user provided a string, convert to callable
+        if isinstance(func, str):
+            if func == 'max':
+                func = np.amax
+            elif func == 'mean':
+                func = np.mean
+            elif func == 'sum':
+                func = np.sum
+            else:
+                raise ValueError('func should be max, mean, or sum')
+        elif callable(func):
+            warnings.warn('User-specific functions are not logged')
+        self.func = func
+
+    _DEFAULT_TESTING_PARAMETERS = {"dims": 'r', "func": 'max'}
+
+    def run(
+            self,
+            stack: ImageStack,
+            in_place: bool = False,
+            verbose: bool = False,
+            n_processes: Optional[int] = None,
+            *args,
+    ) -> ImageStack:
+        """Performs the dimension reduction with the specifed function
+
+        Parameters
+        ----------
+        stack : ImageStack
+            Stack to be filtered.
+        in_place : bool
+            if True, process ImageStack in-place, otherwise return a new stack
+        verbose : bool
+            if True, report on filtering progress (default = False)
+        n_processes : Optional[int]
+            Number of parallel processes to devote to calculating the filter
+
+        Returns
+        -------
+        ImageStack :
+            If in-place is False, return the results of filter as a new stack. Otherwise return the
+            original stack.
+
+        """
+        if not in_place:
+            stack = deepcopy(stack)
+
+        # Apply the reducing function
+        reduced = stack._data.reduce(self.func, dim=[dim.value for dim in self.dims])
+
+        # Add the reduced dims back and align with the original stack
+        reduced = reduced.expand_dims(tuple(dim.value for dim in self.dims))
+        reduced = reduced.transpose(*stack.xarray.dims)
```

Comment:
I wonder if the physical coordinates are correct here.  You may want to look at #1291.

Reply:
Great point. Shannon said she would merge that soon and I can fix the physical coordinates.

Reply:
I just landed #1353, which encompasses #1291

Reply:
Awesome! Thanks @ttung and @shanaxel42 !